src/blk/kernel/KernelDevice.cc

```diff
@@ -94,9 +94,13 @@ int KernelDevice::_lock()
   int fd = fd_directs[WRITE_LIFE_NOT_SET];
   uint64_t nr_tries = 0;
   for (;;) {
+#if defined(F_OFD_SETLK)
```

Comment:
how about
```c++
#ifndef F_OFD_SETLK
#define F_OFD_SETLK F_SETLK
#endif
```
?

in `compact.h` ? so on distro/systems without open file description lock, POSIX record lock is used.

Reply:
@tchaikov 
It wil still need a change in this file, since the order of components in the struc flock is different. 
Which shows the danger of just setting a struct with ` = { ........ }`
So I'll try and fix both for FreeBSD.

```
FreeBSD:
struct flock {
	     off_t   l_start;	     /*	starting offset	*/
	     off_t   l_len;	     /*	len = 0	means until end	of file	*/
	     pid_t   l_pid;	     /*	lock owner */
	     short   l_type;	     /*	lock type: read/write, etc. */
	     short   l_whence;	     /*	type of	l_start	*/
	     int     l_sysid;	     /*	remote system id or zero for local */
     };

Linux:
 struct flock {
               ...
               short l_type;    /* Type of lock: F_RDLCK,
                                   F_WRLCK, F_UNLCK */
               short l_whence;  /* How to interpret l_start:
                                   SEEK_SET, SEEK_CUR, SEEK_END */
               off_t l_start;   /* Starting offset for lock */
               off_t l_len;     /* Number of bytes to lock */
               pid_t l_pid;     /* PID of process blocking our lock
                                   (set by F_GETLK and F_OFD_GETLK) */
               ...
           };
```